src/Model/Behavior/UploadBehavior.php

```diff
@@ -187,12 +187,12 @@ public function getPathProcessor(EntityInterface $entity, $data, string $field,
      * Retrieves an instance of a file writer which knows how to write files to disk
      *
      * @param \Cake\Datasource\EntityInterface $entity an entity
-     * @param array $data the data being submitted for a save
+     * @param \Zend\Diactoros\UploadedFile $data the data being submitted for a save
      * @param string $field the field for which data will be saved
      * @param array $settings the settings for the current field
      * @return \Josegonzalez\Upload\File\Writer\WriterInterface
      */
-    public function getWriter(EntityInterface $entity, array $data, string $field, array $settings): WriterInterface
+    public function getWriter(EntityInterface $entity, UploadedFile $data, string $field, array $settings): WriterInterface
```

Comment:
I'm fixed some tests but i dont know witch is the best approach on param ``$data`` in case of ``afterDelete`` callback... convert the filename stored on database to a ``UploadedFileInterface`` concrete object? I don't see much sense in that. The same with ``getPathProcessor``

Reply:
I prefer object or null personally, so that I don't need to interrogate the object to see if it's empty, and I can use a nullable type hint. Not sure what other folks like 😄